### PR TITLE
fix: add `git_version` to both chunk and batch proofs

### PR DESCRIPTION
**Pull request by silathdiir**
### Summary

- Add a function `short_git_version` to get short commit version.
- Add this git version to both chunk and batch proofs.

### CODE REVIEW

**Review by lispc — DISMISSED** [review]

**Comment by lispc** [comment]
can you give some example of how the git version string likes like?

File: prover/src/proof/batch.rs [hunk 1]
```diff
@@ -32,11 +33,14 @@ impl From<Proof> for BatchProof {
...
         let instances = serialize_instance(&instances[0][ACC_LEN..]);
 
+        let git_version = Some(GIT_VERSION.to_string());
+
         Self {
             raw: Proof {
                 proof,
                 instances,
                 vk,
+                git_version,
```
Comment (icemelon) [inline prover/src/proof/batch.rs:43]:
```suggestion
                proof.git_version,
```

File: prover/src/proof/batch.rs [hunk 2]
```diff
@@ -32,11 +33,14 @@ impl From<Proof> for BatchProof {
         // raw_instances = pi_data
         let instances = serialize_instance(&instances[0][ACC_LEN..]);
 
+        let git_version = Some(GIT_VERSION.to_string());
```
Comment (icemelon) [inline prover/src/proof/batch.rs:36]: remove

**Review by icemelon — DISMISSED** [review]

**Comment by silathdiir** [comment]
@HAOYUatHZ I use this command to generate short git version (7-chars).
```
git rev-parse --short HEAD
```

**Comment by HAOYUatHZ** [comment]
please don't merge now. having some discussions with @silathdiir

**Review by icemelon — APPROVED** [review]

**Pull request merged**